misc-scripts/xref_mapping/XrefParser/RefSeqCoordinateParser.pm

```diff
@@ -23,405 +23,300 @@ use strict;
-
-  } elsif (defined $db) {
+    $registry->load_registry_from_db(
+      '-host' => $file_params->{ofhost},
+      '-port' => $file_params->{ofport},
+      '-user' => $file_params->{ofuser},
+    );
+    $otherf_dba = $registry->get_DBAdaptor($species_name, 'otherfeatures');
+    $otherf_dba->dnadb($core_dba);
+  # for ensemblgenomes project, use staging and ignore any connection details provided
+  } elsif ( $file_params->{project} eq ('ensemblgenomes') ) {
+    $registry->load_registry_from_multiple_dbs( {
+      '-host' => 'mysql-eg-staging-1.ebi.ac.uk',
+      '-port' => '4160',
+      '-user' => 'ensro',
+    }, {
+      '-host' => 'mysql-eg-staging-2.ebi.ac.uk',
+      '-port' => '4275',
+      '-user' => 'ensro',
+    } );
+    $core_dba = $registry->get_DBAdaptor($species_name, 'core');
+    $otherf_dba = $registry->get_DBAdaptor($species_name, 'otherfeatures');
+  # if no project but db provided, use that
+  } else {
     $otherf_dba = $db;
     $core_dba = $db->dnadb();
-  } else {
-      die("Missing or unsupported project value. Supported values: ensembl, ensemblgenomes");
   }
 
-## Not all species have an otherfeatures database, skip if not found
+  # Not all species have an otherfeatures database, skip if not found
   if (!$otherf_dba) {
-    print STDERR "No otherfeatures database for $species_name, skipping import for refseq_import data\n";
-    return;
+    croak "No otherfeatures database for $species_name, skipping import for refseq_import data\n";
   }
 
-## Add link to EntrezGene IDs where available
-  my (%entrez_ids) = %{ $self->get_valid_codes("EntrezGene", $species_id, $dbi) };
+  # Add link to EntrezGene IDs where available
+  my $entrez_ids = $self->get_valid_codes("EntrezGene", $species_id, $dbi);
+
   my $entrez_source_id = $self->get_source_id_for_source_name('EntrezGene', undef, $dbi);
-  my $add_dependent_xref_sth = $dbi->prepare("INSERT INTO dependent_xref  (master_xref_id,dependent_xref_id, linkage_source_id) VALUES (?,?, $entrez_source_id)");
 
   my $sa = $core_dba->get_SliceAdaptor();
   my $sa_of = $otherf_dba->get_SliceAdaptor();
   my $chromosomes_of = $sa_of->fetch_all('toplevel', undef, 1);
 
-# Fetch analysis object for refseq
+  # Fetch analysis object for refseq
   my $aa_of = $otherf_dba->get_AnalysisAdaptor();
-  my $logic_name;
-  foreach my $ana(@{ $aa_of->fetch_all() }) {
-    if ($ana->logic_name =~ /refseq_import/) {
-      $logic_name = $ana->logic_name;
-    }
-  }
-## Not all species have refseq_import data, skip if not found
-  if (!defined $logic_name) {
-    print STDERR "No data found for RefSeq_import, skipping import\n";;
-    return;
+
+  # Not all species have refseq_import data, skip if not found
+  if (!defined $aa_of->fetch_by_logic_name('refseq_import')->logic_name) {
+    croak "No data found for RefSeq_import, skipping import\n";
   }
 
-  foreach my $chromosome_of (@$chromosomes_of) {
+  # start main loop
```

Comment:
Would you annotate this "Iterate over chromosomes in otherfeatures database" or something similar? It may not be obvious what '_of' stands for.